File: 251115/정렬된 수 위치 알아내기/indices-of-sorted-array.py
n = int(input())
sequence = list(map(int, input().split()))

class Number:
    def __init__(self, n, idx, after):
        self.n = n
        self.idx = idx
        self.after = after

sequences = []
for i in range(n):
    sequences.append(Number(sequence[i], i + 1, 0))

sequences.sort(key=lambda x: x.n)

for i in range(n):
    sequences[i].after = i + 1

sequences.sort(key=lambda x: x.idx)

for seq in sequences:
    print(seq.after, end=' ')

File: 260225/정렬된 수 위치 알아내기/indices-of-sorted-array.js
class Sequence {
    constructor(number, index) {
        this.number = number;
        this.index = index;
    }
}

const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const arr = input[1].split(' ').map(Number);

let numbers = []
for (let i = 0; i < arr.length; i++) {
    numbers.push(new Sequence(arr[i], i));
}


numbers.sort((a, b) => {
    if (a.number === b.number) return a.index - b.index; // 순서 보장
    return a.number - b.number;
})

let answer = []; // 정답 배열
for (let i = 0; i < arr.length; i++) {
    answer[numbers[i].index] = i + 1; // 이동한 만큼이니까 + 1해줌
}

console.log(answer.join(' '));